lxd/instance/instance_utils.go

```diff
@@ -1175,3 +1175,68 @@ func NextSnapshotName(s *state.State, inst Instance, defaultPattern string) (str
 
 	return pattern, nil
 }
+
+// GetMigratableInstances returns a list of instances which can be migrated.
+func GetMigratableInstances(cluster *db.Cluster, instances []Instance) []Instance {
+	ret := make([]Instance, 0)
+
+	// Check if all migrations can be performed. If not, fail here.
+	for _, inst := range instances {
+		config := inst.ExpandedConfig()
+
+		val, ok := config["cluster.evacuate"]
+		if !ok {
+			val = "auto"
+		}
+
+		if val == "migrate" {
+			ret = append(ret, inst)
+		}
+
+		if val == "stop" {
+			continue
+		}
+
+		devices := inst.ExpandedDevices()
+
+		canMigrate := true
+
+		for _, dev := range devices {
```

Comment:
@stgraber @monstermunchkin this feels like we are starting to leak device specific config back into the instances package.

Shouldn't this be a property of the device?

Reply:
Yeah, a Migratable() function would be nice to do this cleanly.

Reply:
> Yeah, a Migratable() function would be nice to do this cleanly.

Yeah, although I was thinking more an `Inf0()` function like the `Instance` interface, that returns a struct that we could later extend with additional properties if needed. But for now could just have a `Migratable` bool field.

Reply:
Ah yeah, that'd work.

@monstermunchkin can you add an `Info()` function to the `Device` interface and have that return a `DeviceInfo` struct which has a `Migratable` bool? Then update the logic in `driver_common.go` to rely on that rather than the hardcoded device types and details to decide whether an instance is migratable.